nibabel/freesurfer/io.py

```diff
@@ -45,35 +45,20 @@ def _fread3_many(fobj, n):
     return (b1 << 16) + (b2 << 8) + b3
 
 
-def _read_volume_info(extra):
+def _read_volume_info(fobj):
     volume_info = OrderedDict()
-
-    if extra is None:
-        return volume_info
-
-    if extra[:4] != b'\x00\x00\x00\x14':
+    head = np.fromfile(fobj, '>i4', 3)
+    if any(head != [2, 0, 20]):
         warnings.warn("Unknown extension code.")
     else:
-        try:
-            for line in extra[4:].split(b'\n'):
-                if len(line) == 0:
-                    continue
-                key, val = map(bytes.strip, line.split(b'=', 1))
-                key = key.decode('utf-8')
-                if key in ('voxelsize', 'xras', 'yras', 'zras', 'cras'):
-                    val = np.fromstring(val, sep=' ')
-                    val = val.astype(np.float)
-                elif key == 'volume':
-                    val = np.fromstring(val, sep=' ', dtype=np.uint)
-                    val = val.astype(np.int)
-                volume_info[key] = val
-        except ValueError:
-            raise ValueError("Error parsing volume info")
-
-    if len(volume_info) == 0:
-        warnings.warn("Volume geometry info is either "
-                      "not contained or not valid.")
-
+        volume_info['head'] = head
+        for key in ['valid', 'filename', 'volume', 'voxelsize', 'xras', 'yras',
+                    'zras', 'cras']:
+            pair = fobj.readline().split('=')
+            if pair[0].strip() != key or len(pair) != 2:
+                raise IOError('Error parsing volume info.')
+            volume_info[pair[0]] = pair[1]
```

Comment:
Personal preference: I'd rather `volume_info` use keys without spaces, and then use `.ljust(6)` to get the old spacing back on write. If I wanted to pull out xras, I'd find it more intuitive to do `volume_info['xras']`, not `volume_info['xras  ']`.